src/AzureReadStream.cs

```diff
@@ -69,8 +70,15 @@ public override async Task<int> ReadAsync(byte[] buffer, int offset, int count,
 
             if (count > 0)
             {
+                var ms = new MemoryStream();
                 await AzureRetry.Do(
-                    c => _blob.DownloadRangeToByteArrayAsync(buffer, offset, position, count, null, null, null, c),
+                    async c =>
+                    {
+                        using (var s = await _blob.OpenReadAsync(offset, count))
```

Comment:
The reason for this class `Lokad.ContentAddr.Azure.AzureReadStream` is that the blob stream class returned by `_blob.OpenReadAsync` did not (in the old version of the Azure client library) provide any guarantees about how many bytes are read, or how many API requests are made. 

Can you please confirm that in the current version of the Azure client library, the code below will make exactly one request, downloading exactly 6MB of data ? 

```
const int offset = 1024;
const int count = 6 * 1024 * 1024;
using (var s = await _blob.OpenReadAsync(offset, count))
{
    await s.ReadAsync(buffer, offset, count);
}
```

My worry is that one of the following may happen: 

 - `OpenReadAsync` makes its own request to check the blob's existence or query the blob size.
 - `ReadAsync` uses 4MB blocks, which means two requests and 8MB downloaded when only 6MB were needed.

Reply:
You are correct, the current version of the SDK is sending 2 requests, I've attached screenshot of the Fiddler Debugger from my machine.

1- HEAD request for existence (req. \#18)
2- GET request downloading only 6MB (req. \#19)

![image](https://user-images.githubusercontent.com/1621149/169089565-5c604d81-fb68-421e-b58f-02d11aa3c06f.png)

Reply:
@VictorNicollet just wanted to make sure you have seen this, and that it's OK to have the behavior shown in the provided screenshot.

Reply:
@EslaMx7 Thank you ! We should get rid of the HEAD request, as it adds unnecessary latency.

Reply:
@EslaMx7 A solution which might work: to produce a SAS Uri and use an HttpClient to make the request, using the `Range` or `x-ms-range` header as described in the [REST API](https://docs.microsoft.com/en-us/rest/api/storageservices/specifying-the-range-header-for-blob-service-operations).

Reply:
@VictorNicollet Nice Hack! let me give it a try, and let you know, I hope it works, and generating SAS Uri doesn't send any requests also.